examples/asr/run_helper.py

```diff
@@ -80,38 +120,138 @@ def check_missing_values(cfg):
                     raise ValueError(f"Missing value for key {k} in the config file")
 
     check_missing_values(result)
+
+    # Do name check as a special case
+    if 'name' in result and result['name'] == '':
+        raise ValueError(
+            f"Missing value for key 'name' in the merged config file (value={result['name']}).\n"
+            f"Check if your ++ override is using single quote (') instead of double quote (\") for resolution.\n"
+            "Example: ++name='${exp_name}'"
+        )
+
     return result
 
 
 def check_config_mount_paths(script_config, cluster_config):
+    """
+    Check if all path-like strings in the script config are mounted paths in the cluster config.
+    If a path-like string is not a mounted path, raise an error.
+
+    Args:
+        script_config: Script config dictionary that represents the Model training/inference config
+        cluster_config: Cluster config dictionary that represents the cluster configuration
+    """
     # recursively walk all values of the script_config, checking if its a path-like string and if so, check if the path is a mounted path
     # if it is not, raise an error
 
+    if 'AIS_ENDPOINT' in os.environ:
+        ais_endpoint = os.environ['AIS_ENDPOINT']
+    else:
+        ais_endpoint = None
+
+    # Check if ais paths should be checked at all
+    # This can be disabled using `++check_ais_paths=False` passed when calling the script
+    check_ais_paths = cluster_config.get('check_ais_paths', True)
+
+    def filepath_check(v, cluster_cfg):
+        if v.startswith(os.path.sep):  # check for absolute paths only
+            logging.info(f"Checking if {v} is a mounted path")
+            # Check if the path begins with mount path
+            run_utils.check_if_mounted(cluster_cfg, v)
+
+            # Check the file exists in the cluster at the unmounted path
+            unmounted_path = run_utils.get_unmounted_filepath(cluster_cfg, v)
+            run_utils.check_remote_mount_directories(unmounted_path, cluster_cfg)
+
+        elif (
+            check_ais_paths and "ais://" in v and ais_endpoint is not None
```

Comment:
In case my earlier comment was missed -- re-iterating that AIStore handles all kinds of URL schemas, the condition that leads to AIStore code branch here should:
1) check whether a path is an URL/URI
2) check whether `ais_endpoint is not None`